docs/sources/reference/builder.md

```diff
@@ -350,8 +350,9 @@ accessible from the host by default. To expose ports to the host, at runtime,
 
 The `ENV` instruction sets the environment variable `<key>` to the value
 `<value>`. This value will be passed to all future 
-`RUN`, `ENTRYPOINT`, and `CMD` instructions. This is
-functionally equivalent to prefixing the command with `<key>=<value>`
+`RUN`, `ENTRYPOINT`, and `CMD` instructions. To set a value for a single
+command without affecting future commands or containers, prefix the command
+with `<key>=<value>`.
```

Comment:
@duglin It may not be obvious to everyone that you can set a value in the environment just for a single command. And once you set an `ENV` you can't really unset it. So I still think `RUN foo=bar do-something` should exist. But maybe it doesn't belong here. Maybe it belongs in the [best practices](https://github.com/docker/docker/blob/master/docs/sources/articles/dockerfile_best-practices.md).

Also, I realized the _Environment Replacement_ section starting on line 108 covers your original question. Maybe it just needs to be better tied to the actual `ENV` documentation.